torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1069,6 +1069,39 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
         return inputs
 
 
+def sample_inputs_batch_norm(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    make_arg_without_grad = partial(make_tensor, device=device, dtype=dtype, requires_grad=False)
+
+    # Ordered as: input shape, kwargs for training, momentum, eps
+    cases: Tuple[Tuple[int], dict] = (  # type: ignore[assignment]
+        ((S, S, S), {'training': True, 'momentum': 0.5, 'eps': 0.6}),
+        ((3, 2, 4), {'training': False, 'momentum': -1.2}),
+        ((3, 1), {'training': True, 'momentum': 0.0}),
+        ((2, 1), {'training': False}),
+        ((3, 2, 3, 4), {'training': True, 'momentum': -1.0, 'eps': 0.5}),
+        ((3, 2, 3, 4), {'training': False, 'momentum': -1.0, 'eps': 0.5}),
+    )
+
+    def generator():
+        for input_shape, kwargs in cases:
+            # args: running mean, running var, weight and bias should necessarily be of shape: (channels,)
+            channels = input_shape[1]
+            yield SampleInput(
+                make_arg(input_shape),
+                args=(
+                    make_arg_without_grad(channels, low=0),  # running mean
+                    make_arg_without_grad(channels, low=0),  # running var
+                    make_arg(channels),  # weight
+                    make_arg(channels)   # bias
+                ),
+                kwargs=kwargs
+            )
+        # Test case for no optional kwargs
+        yield SampleInput(make_arg((1, 2, 3)), args=(make_arg_without_grad(2, low=0),
+                                                     make_arg_without_grad(2, low=0)))
```

Comment:
It's not well documented, but it turns out `running_mean` and `running_var` are allowed to be None. Could we add a test case for that as well? (I'm not sure if they both have to be none or if it is fine for one of them to be not-none, we should run the code and see...)